tools/scalajsenv.js

```diff
@@ -570,36 +570,36 @@ ScalaJS.uJ = function(value) {
 
 // TypeArray conversions
 
-ScalaJS.byteArray2TypedArray = function(value) { return new Int8Array(value.u); };
-ScalaJS.shortArray2TypedArray = function(value) { return new Int16Array(value.u); };
-ScalaJS.charArray2TypedArray = function(value) { return new Uint16Array(value.u); };
-ScalaJS.intArray2TypedArray = function(value) { return new Int32Array(value.u); };
-ScalaJS.floatArray2TypedArray = function(value) { return new Float32Array(value.u); };
-ScalaJS.doubleArray2TypedArray = function(value) { return new Float64Array(value.u); };
+ScalaJS.byteArray2TypedArray = function(value) { return new ScalaJS.g["Int8Array"](value.u); };
+ScalaJS.shortArray2TypedArray = function(value) { return new ScalaJS.g["Int16Array"](value.u); };
+ScalaJS.charArray2TypedArray = function(value) { return new ScalaJS.g["Uint16Array"](value.u); };
+ScalaJS.intArray2TypedArray = function(value) { return new ScalaJS.g["Int32Array"](value.u); };
+ScalaJS.floatArray2TypedArray = function(value) { return new ScalaJS.g["Float32Array"](value.u); };
+ScalaJS.doubleArray2TypedArray = function(value) { return new ScalaJS.g["Float64Array"](value.u); };
```

Comment:
In theory, yes. In practice, this would be premature optimization compared to regular usages of Typed Arrays in Scala.js code: those will be emitted as `ScalaJS.g["Int8Array"]` anyway (or `$g["Int8Array"]` for the modern `Isolated` translation).